src/system-health/health_checker/sysmonitor.py

```diff
@@ -0,0 +1,634 @@
+            (state, c) = sel.select(SELECT_TIMEOUT_MSECS)
+            if state == swsscommon.Select.TIMEOUT:
+                continue
+            if state != swsscommon.Select.OBJECT:
+                logger.log_warning("sel.select() did not return swsscommon.Select.OBJECT")
+                continue
+            (key, op, cfvs) = cst.pop()
+            key_ext = key+".service"
+            #logger.log_info("featureevent received:{}".format(key_ext))
+            timestamp="{}".format(datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"))
+            msg={"unit": key_ext,"evt_src":"feature","time":timestamp}
+            self.task_notify(msg)
+            time.sleep(0.1)
+        
+        #logger.log_info("Stop STATE_DB monitoring loop")
+        
+    def task_worker(self):
+        while not self.task_stopping_event.is_set():
+            try:
+                self.task_worker_statedb()
+            except Exception as e:
+                logger.log_error("task_worker_statdb exited- {}, retry".format(str(e)))
+            time.sleep(1)
+    
+
+    def task_run(self):
+        if self.task_stopping_event.is_set():
+            return
+        self.task_process = multiprocessing.Process(target=self.task_worker)
+        self.task_process.start()
+
+    def task_notify(self, msg):
+        if self.task_stopping_event.is_set():
+            return
+        self.task_queue.put(msg)
+        time.sleep(0.001)
+
+    def task_stop(self):
+        self.task_stopping_event.set()
+        self.task_process.join()
+
+
+class MonitorSystemBusTask(object):
+    def __init__(self):
+        self.task_queue = mpmgr.Queue()
+        self.task_process = None
+        self.task_stopping_event = multiprocessing.Event()
+
+    def __init__(self,myQ):
+        self.task_queue = myQ
+        self.task_process = None
+        self.task_stopping_event = multiprocessing.Event()
+
+    def OnJobRemoved(self, id, job, unit, result):
+        #print('{}: Job Removed: {}, {}, {} '.format( id, job, unit, result))
+        if result == "done":
+            #logger.log_info("Received systemd service event for {}".format(unit))
+            timestamp="{}".format(datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"))
+            msg={"unit": unit,"evt_src":"sysbus","time":timestamp}
+            self.task_notify(msg)
+            return
+
+    #Sub process for listening the systemd event on dbus
+    def subscribe_sysbus(self):
+        import dbus
+        from gi.repository import GObject
+        from dbus.mainloop.glib import DBusGMainLoop
+
+        DBusGMainLoop(set_as_default=True)
+
+        bus = dbus.SystemBus()
+        systemd = bus.get_object('org.freedesktop.systemd1', '/org/freedesktop/systemd1')
+        manager = dbus.Interface(systemd, 'org.freedesktop.systemd1.Manager')
+
+        manager.Subscribe()
+        manager.connect_to_signal('JobRemoved', self.OnJobRemoved)
+
+        loop = GObject.MainLoop()
+        loop.run()
```

Comment:
is loop.run() blocking or non-blocking?

Reply:
https://dbus.freedesktop.org/doc/dbus-python/tutorial.html#making-asynchronous-method-calls